frontend/src/metabase/visualizations/types/visualization.ts

```diff
@@ -75,56 +72,154 @@ export interface VisualizationProps {
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => TValue;
   persistDefault?: boolean;
+
+  getValue?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => TValue;
+  isValid?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => boolean;
+
+  section?: string;
+  getSection?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => string;
+
+  title?: string;
+  getTitle?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => string;
+
+  hidden?: boolean;
+  getHidden?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => boolean;
+
+  marginBottom?: string;
+  getMarginBottom?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => string;
+
+  disabled?: boolean;
+  getDisabled?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    extra: TExtra,
+  ) => boolean;
+
   props?: TProps;
   getProps?: (
-    series: Series,
-    vizSettings: VisualizationSettings,
+    object: TObject,
+    settings: VisualizationSettings,
     onChange: (value: TValue) => void,
-    extra: unknown,
+    extra: TExtra,
   ) => TProps;
-  readDependencies?: string[];
-  writeDependencies?: string[];
-  eraseDependencies?: string[];
-  // is the setting visible in the dashboard card viz settings
-  dashboard?: boolean;
-  useRawSeries?: boolean;
+  getExtraProps?: (
+    object: TObject,
+    settings: VisualizationSettings,
+    onChange: (value: TValue) => void,
+    extra: TExtra,
+  ) => TProps;
+  onUpdate?: (value: TValue, extra: TExtra) => void;
 };
 
-export type VisualizationSettingsDefinitions = {
-  [key: string]: VisualizationSettingDefinition<unknown, unknown>;
+export type VisualizationSettingsDefinitions<TObject = unknown> = {
+  [id in VisualizationSettingId]?: VisualizationSettingDefinition<
+    TObject,
+    VisualizationSettings[id]
+  >;
 };
 
+export type VisualizationColumnSettingsDefinitions<TObject = unknown> = {
+  [id in VisualizationColumnSettingsId]?: VisualizationSettingDefinition<
+    TObject,
+    VisualizationColumnSettings[id]
+  >;
+};
+
+// returned by getSettingWidget
+export type VisualizationSettingWidget<TObject = unknown, TValue = any> = {
+  id: string;
+  value: TValue;
+  set: boolean;
+  widget: any;
+  onChange: (newValue: TValue, question: Question) => void;
+  onChangeSettings: (
+    newSettings: VisualizationSettings,
+    question: Question,
+  ) => void;
+} & VisualizationSettingDefinition<TObject, TValue>;
+
 export type VisualizationGridSize = {
   // grid columns
   width: number;
   // grid rows
   height: number;
 };
 
-// TODO: add component property for the react component instead of the intersection
-export type Visualization = React.ComponentType<VisualizationProps> & {
-  name: string;
-  noun: string;
+export type VisualizationProperties = {
```

Comment:
`VisualizationProperties` is a slightly ambiguous name, I think just `Visualization` looked fine to me